core/src/main/scala/org/apache/spark/api/r/RBackendHandler.scala

```diff
@@ -143,12 +142,8 @@ private[r] class RBackendHandler(server: RBackend)
       val cls = if (isStatic) {
         Utils.classForName(objId)
       } else {
-        JVMObjectTracker.get(objId) match {
-          case None => throw new IllegalArgumentException("Object not found " + objId)
-          case Some(o) =>
-            obj = o
-            o.getClass
-        }
+        obj = server.jvmObjectTracker.get(JVMObjectId(objId))
```

Comment:
The `get` returns `null` but the next line will throw the exception. The old code was instead throwing a `IllegalArgumentException("Object not found " + objId)`